sdk_contrib/fetch/lib/fetch_p.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * @module fetch_p
+ */
+
+/**
+ * This module patches the global fetch instance for NodeJS 18+
+ */
+
+const contextUtils = require('aws-xray-sdk-core/lib/context_utils');
+const utils = require('aws-xray-sdk-core/lib/utils');
+const logger = require('aws-xray-sdk-core/lib/logger');
+
+/**
+ * Wrap fetch either in global instance for recent NodeJS or the node-fetch module for older versions,
+ *  to automatically capture information for the segment.
+ * This patches the built-in fetch function globally.
+ * @param {boolean} downstreamXRayEnabled - when true, adds a "traced:true" property to the subsegment
+ *   so the AWS X-Ray service expects a corresponding segment from the downstream service.
+ * @param {function} subsegmentCallback - a callback that is called with the subsegment, the fetch request,
+ *   the fetch response and any error issued, allowing custom annotations and metadata to be added.
+ * @alias module:fetch_p.captureFetch
+ */
+function captureFetch(downstreamXRayEnabled, subsegmentCallback) {
```

Comment:
This function seems unneeded, may introduce confusion. Without looking in this captureFetch function, a user may not understand if they are capturing the Fetch module or the Fetch in the global instance. A user would probably have an idea which one they want to capture explicity, so I think just keeping `captureFetchModule` and `captureFetchGlobal` would be fine. Otherwise, the version of Fetch that is actually instrumented is a bit abstracted.

Reply:
Abstraction was kind of the point :) but nonetheless, I've removed it and updated unit tests and documentation